src/vcpkg/postbuildlint.cpp

```diff
@@ -346,10 +353,16 @@ namespace vcpkg::PostBuildLint
     {
         const auto packages_dir = paths.packages() / spec.dir();
         const auto copyright_file = packages_dir / "share" / spec.name() / "copyright";
-        if (fs.exists(copyright_file, IgnoreErrors{}))
+        if (fs.is_regular_file(copyright_file))
         {
             return LintStatus::SUCCESS;
         }
+        else if (fs.is_directory(copyright_file))
+        {
+            msg::println_warning(msgCopyrightIsDir, msg::path = "copyright");
```

Comment:
I think this should return `return LintStatus::PROBLEM_DETECTED;`? It does mean that we need to fix the existing curated catalog to comply though.

Reply:
`LintStatus::PROBLEM_DETECTED` turns into a warning for `vcpkg install` customers, but an error in our CI, so this is consistent with https://github.com/microsoft/vcpkg/pull/19227#issuecomment-912005333